bgpd/bgp_main.c

```diff
@@ -82,6 +82,8 @@ static const struct option longopts[] = {
 	{"socket_size", required_argument, NULL, 's'},
 	{0}};
 
+DEFINE_HOOK(bgp_snmp_shutdown, (), ());
```

Comment:
Nit: hooks should be named by their position / "always" function in the containing code, rather than the one random use case they were introduced for. Nothing about this hook is SNMP specific, and some other module that needs to do thinks on teardown may want to use the hook as well — and then the name would get confusing.